lxd/daemon_storage.go

```diff
@@ -76,6 +76,24 @@ func daemonStorageUnmount(s *state.State) error {
 		}
 	}
 
+	pools, err := s.Cluster.GetStoragePoolNames()
+	if err != nil {
+		return errors.Wrap(err, "Failed to get storage pools")
+	}
+
+	for _, poolName := range pools {
+		pool, err := storagePools.GetPoolByName(s, poolName)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to get storage pool %q", poolName)
```

Comment:
@masnax a recent change @stgraber and I were discussing was around stopping using the `github.com/pkg/errors` package and switching to the Go stdlib `errors` package for error wrapping.

So whilst this switch over will take a while (as we're not doing it in one go big-bang), I would recommend avoiding using it for new code.

So here can you use this pattern instead please:

```go
return fmt.Errorf("Failed to get storage pool %q: %w", poolName, err)
```

For more info see https://go.dev/blog/go1.13-errors

We've already updated the `SmartError()` function to detect specific errors wrapped using this approach so should be safe to use everywhere now, along with the new `api.StatusErrorf()` function for controlling the HTTP status code returned via an error.